src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2444,14 +2444,24 @@ private bool InstructionSelfDestruct<TTracing>(EvmState vmState, ref EvmStack<TT
         }
 
         _state.SubtractFromBalance(env.ExecutingAccount, value, spec);
+
+        ValueKeccak codeHash = ValueKeccak.Compute(initCode);
```

Comment:
This is code that was just loaded from memory and not state, why do you think it is likely it lives in cache? And why would it be reused?

Reply:
128 create2 contracts for CoinTool: XEN Batch Miner in this recent block; code identical https://etherscan.io/txsinternal?ps=100&zero=false&block=17534199&valid=all

Lot of `.ToArray()` on contract sized memory chunks if code already in cache (or could be added to cache before second tx)